### DISCUSSION TITLE
Guidance for powering pi zero with buck converter?

### DISCUSSION THREAD

**Discussion by willheineman** [aqualinkd/AqualinkD#255] [General]
Does anyone have any basic guidance for powering the pi-zero from the RS485 using a buck converter? This is alluded to in the Wiki, but was never fully written out.

https://github.com/sfeakes/AqualinkD/wiki#advanced

**Comment by Bugman1400** [comment]
This seems self explanatory.
![image](https://github.com/sfeakes/AqualinkD/assets/25798328/15ee8bd9-a4e8-4efa-8515-ade11ab3d09c)

**Reply by sjames99** [reply]
They might be looking for guidance on where to solder the wires.

**Reply by willheineman** [reply]
Yup, I was hoping someone had a quick picture or diagram showing where its beat to pull the input from on the RS485. I'll figure it out if nobody has anything, but I was hoping it existed already.

**Reply by sfeakes** [reply]
It’s hard to give a picture as wire colours will change depending on install.  
For power, go to the control panel and find the RS485 connection, then look at the wire colours there.  There are 4 wires.  Ground, Data +, Data -, Positive.  (Labeled 4,3,2,1 in that order). You want to connect the Positive and Ground to the power converter, and the Data + and Data - to the USB2RS485 converter.   You would usually do this inside, but you need the correct wire colours  from the control panel.  IF (and that’s a big if) your pool insulation followed the recommended wire colours, these are Red and Green.


The output of your power converter then goes to the pi for power.  Micro usb connection would be the best, but you can also backpower the pi from the GPIO. While back powering  is easier it’s not recommended since that will bypass any over voltage protection the Pi has.

so your power converter input is connected to the Power and Ground of the RS485 bus and the Output to a microUSB plug.

**Comment by lodged78** [comment]
The problem with a buck converter is the RS485 power supply is unstable, especially if connecting at the interior control panel. My VCC is only 9v at the interior panel, and my Pi Zero frequently locked up with a straight buck converter.

Easiest solution is power stabilizer like the PiSugar 3, it has a USB-C power input. Buy a buck converter with a USB-C output; the red wire goes to RS485 terminal 1 (VCC), and the black wire to RS485 terminal 4 (GND).

The PiSugar will prevent your SD Card from being corrupted by power outages.

https://www.pisugar.com/
You can buy on Amazon. It's open source and includes STL files to print a nice case. I'm going to create a larger case to hold all the wires inside one unit at some point - I'll post it when I do.


> Does anyone have any basic guidance for powering the pi-zero from the RS485 using a buck converter? This is alluded to in the Wiki, but was never fully written out.
> 
> https://github.com/sfeakes/AqualinkD/wiki#advanced

**Comment by chrisdaun** [comment]
@willheineman 

Here is a write up on my setup using a dc-to-dc converter powered off the bus voltage.

https://docs.google.com/document/d/1oaTGzKCac2eOb3nlkELobyPfplSXrlZATgusxWKnMqk

So far I have had no issues.
Hope this helps :)

![image](https://github.com/sfeakes/AqualinkD/assets/1088495/1c234271-0723-41da-9618-528e5975be6e)
